frontend/src/component/common/NewConstraintAccordion/ConstraintsList/ViewableConstraintsList.tsx

```diff
@@ -27,9 +27,9 @@ export const ViewableConstraintsList = ({
     return (
         <StyledContainer>
             <ConstraintsList>
-                {constraints.map((constraint) => (
+                {constraints.map((constraint, index) => (
                     <ConstraintAccordionView
-                        key={constraint[constraintId]}
+                        key={constraint[constraintId] || index}
```

Comment:
viewable constraints will often not have constraint ids (because they're not intended to be edited). As such, we can fall back to using the index here, because there's no other differentiator.